gst/nnstreamer/tensor_common.c

```diff
@@ -336,11 +386,19 @@ gboolean
 gst_tensors_info_is_equal (const GstTensorsInfo * i1, const GstTensorsInfo * i2)
 {
   guint i;
+  gboolean compatible, flexible;
 
   g_return_val_if_fail (i1 != NULL, FALSE);
   g_return_val_if_fail (i2 != NULL, FALSE);
 
-  if (i1->num_tensors != i2->num_tensors || i1->num_tensors < 1) {
+  flexible = gst_tensors_info_is_flexible (i1);
+  if (flexible != gst_tensors_info_is_flexible (i2)) {
+    return FALSE;
+  }
+
+  compatible = (i1->num_tensors > 0 || i2->num_tensors > 0) || flexible;
```

Comment:
Same result, but may be better for readability.
```suggestion
  compatible = (i1->num_tensors > 0 && i2->num_tensors > 0) || flexible;
```

Reply:
updated